File: TourismAgency-frontend/src/app/model/employee-entity.ts
export class EmployeeEntity {
    idEmployee: number;
    name: string;
    lastName: string;
    address: string;
    dni: string;
    birthdate: string;
    nationality: string;
    cellPhone: string;
    email: string;
    position: string;
    salary: number;

    constructor(
        idEmployee: number = 0,
        name: string = "",
        lastName: string = "",
        address: string = "",
        dni: string = "",
        birthdate: string = "",
        nationality: string = "",
        cellPhone: string = "",
        email: string = "",
        position: string = "",
        salary: number = 0
    ) {
        this.idEmployee = idEmployee;
        this.name = name;
        this.lastName = lastName;
        this.address = address;
        this.dni = dni;
        this.birthdate = birthdate;
        this.nationality = nationality;
        this.cellPhone = cellPhone;
        this.email = email;
        this.position = position;
        this.salary = salary;
    }
}

File: TourismAgency-backend/tourism/src/main/java/com/tourism/model/entity/EmployeeEntity.java
package com.tourism.model.entity;

import java.time.LocalDate;
import java.util.List;

import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import lombok.*;

@Entity
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Table(name = "employee")
@Builder
public class EmployeeEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idEmployee;

    @Column(length = 50, nullable = false)
    private String name;//------KEYCLOAK--------

    @Column(length = 50, nullable = false)
    private String lastName;//------KEYCLOAK--------

    @Column(length = 50, nullable = false)
    private String address;

    @Column(length = 20, nullable = false, unique = true)
    private String dni;

    @Column(nullable = false)
    @Temporal(TemporalType.DATE)
    private LocalDate birthdate;

    @Column(length = 20, nullable = false)
    private String nationality;

    @Column(length = 20, nullable = false)
    private String cellPhone;

    @Column(length = 50, nullable = false)
    @Email
    private String email;//------KEYCLOAK--------

    @Column(length = 50, nullable = false)
    private String position;

    @Column(nullable = false)
    private Double salary;

    //relationships between tables
    @OneToMany(mappedBy = "employee", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<SaleEntity> sales;
}
